src/main/template/vpc.yaml

```diff
@@ -0,0 +1,410 @@
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} NAT Gateway AZ A
+
+  natGatewayB:
+    Type: AWS::EC2::NatGateway
+    Properties:
+      AllocationId: !GetAtt natGatewayBElasticIP.AllocationId
+      SubnetId: !Ref publicSubnetB
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} NAT Gateway AZ B
+
+  natGatewayC:
+    Type: AWS::EC2::NatGateway
+    Properties:
+      AllocationId: !GetAtt natGatewayCElasticIP.AllocationId
+      SubnetId: !Ref publicSubnetC
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} NAT Gateway AZ C
+
+  privateRouteTableA:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref vpc
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Private Route Table A
+
+  privateRouteTableB:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref vpc
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Private Route Table B
+
+  privateRouteTableC:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref vpc
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Private Route Table C
+
+  privateRouteA:
+    Type: AWS::EC2::Route
+    Properties:
+      RouteTableId: !Ref privateRouteTableA
+      DestinationCidrBlock: 0.0.0.0/0
+      NatGatewayId: !Ref natGatewayA
+
+  privateRouteB:
+    Type: AWS::EC2::Route
+    Properties:
+      RouteTableId: !Ref privateRouteTableB
+      DestinationCidrBlock: 0.0.0.0/0
+      NatGatewayId: !Ref natGatewayB
+
+  privateRouteC:
+    Type: AWS::EC2::Route
+    Properties:
+      RouteTableId: !Ref privateRouteTableC
+      DestinationCidrBlock: 0.0.0.0/0
+      NatGatewayId: !Ref natGatewayC
+
+  privateSubnetARouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref privateRouteTableA
+      SubnetId: !Ref privateSubnetA
+
+  privateSubnetBRouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref privateRouteTableB
+      SubnetId: !Ref privateSubnetB
+
+  privateSubnetCRouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref privateRouteTableC
+      SubnetId: !Ref privateSubnetC
+
+  bastionSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupName: !Sub ${AWS::StackName} Bastion Security Group
+      GroupDescription: !Sub ${AWS::StackName} Bastion Security Group
+      VpcId: !Ref vpc
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 22
+        ToPort: 22
+        CidrIp: 0.0.0.0/0
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Bastion Security Group
+
+  bastion:
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId: !Ref bastionAmi
+      KeyName: !Ref keyName
+      InstanceType: !Ref bastionInstanceType
+      NetworkInterfaces:
```

Comment:
[The documentation on the `NetworkInterfaces` property](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-instance.html#aws-properties-ec2-instance-properties) states that
> If this resource has a public IP address and is also in a VPC that is defined in the same template, you must use the DependsOn Attribute to declare a dependency on the VPC-gateway attachment.

Same question: why does this currently work? O_o

Reply:
See above.